### PR TITLE
[extractor/urplay] Support multi-langguage subtitles

### PR DESCRIPTION
**Pull request by hoaluvn**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

The current version only looks for Swedish subtitle. The *pull request* will pick up all available subtitles.

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

### Summary
🐛🌐🎬

Fix subtitle extraction for URPlay videos by handling Persian language code and checking all streamingInfo keys in `yt_dlp/extractor/urplay.py`

> _`parse_lang_code` changed_
> _Persian bypasses conversion_
> _Autumn leaves subtitles_

### Walkthrough
*  Bypass ISO639Utils conversion for Persian language code in `parse_lang_code` function ([link](https://github.com/yt-dlp/yt-dlp/pull/6707/files?diff=unified&w=0#diff-529ee69433a32c0e73aac5daeb4037fb1a4ed98983402df3a63eb5b9ded20638L111-R129))
*  Iterate over all keys in `streamingInfo` dictionary to extract subtitles in different languages ([link](https://github.com/yt-dlp/yt-dlp/pull/6707/files?diff=unified&w=0#diff-529ee69433a32c0e73aac5daeb4037fb1a4ed98983402df3a63eb5b9ded20638L111-R129))

### CODE REVIEW

File: yt_dlp/extractor/urplay.py [hunk 1]
```diff
@@ -108,22 +108,25 @@ def parse_lang_code(code):
             if code is None:
                 return
             lang = code.lower()
-            if not ISO639Utils.long2short(lang):
-                lang = ISO639Utils.short2long(lang)
+            # bypass for Persian
+            if lang != "per":
```
Comment (pukkandan) [inline yt_dlp/extractor/urplay.py:112]: pls explain this

Comment (hoaluvn) [reply yt_dlp/extractor/urplay.py]: Revert this change, and add Persian language to ISO639Utils in next commit

File: yt_dlp/extractor/urplay.py [hunk 2]
```diff
@@ -112,18 +112,19 @@ def parse_lang_code(code):
...
-            if not sttl_url:
-                continue
-            lang = parse_lang_code(lang)
-            if not lang:
-                continue
-            sttl = subtitles.get(lang) or []
-            sttl.append({'ext': k, 'url': sttl_url, })
-            subtitles[lang] = sttl
+        for langpkg in urplayer_data['streamingInfo']:
+            for k, v in urplayer_data['streamingInfo'].get(langpkg).items():
```
Comment (pukkandan) [inline yt_dlp/extractor/urplay.py:116]: Why not:
```suggestion
        for langpkg in urplayer_data['streamingInfo'].values():
            for k, v in langpkg.items():
```

(Change var name. It's not clear what `langpkg` means)

Comment (pukkandan) [reply yt_dlp/extractor/urplay.py]: > (Change var name. It's not clear what langpkg means)

In case u missed.

Comment (hoaluvn) [reply yt_dlp/extractor/urplay.py]: Thanks for the fix

**Pull request closed without merging**